grype-server/pkg/scanner/scanner.go

```diff
@@ -133,12 +135,27 @@ func (s *Scanner) ScanSbomJson(sbom string) (*models.Document, error) {
 }
 
 func (s *Scanner) scan(packagesContext grype_pkg.Context, packages []grype_pkg.Package) (*models.Document, error) {
-	store := store.Store{
+	vulnerabilityMatcher := createVulnerabilityMatcher(store.Store{
 		Provider:          s.vulProvider,
 		MetadataProvider:  s.vulMetadataProvider,
 		ExclusionProvider: s.exclusionProvider,
+	})
+
+	allMatches, ignoredMatches, err := vulnerabilityMatcher.FindMatches(packages, packagesContext)
```

Comment:
before that we passed nil to NewDocument. Can you add a comment why we handle it now (part of this PR)

Reply:
`ignoredMatches` now returned from the new `vulnerabilityMatcher.FindMatches` function.
It is not related to `exclusionProvider`, it is related to `vulnerabilityMatcher.IgnoreRules` (that we are not populating, same as we are not setting `FailSeverity`)
```
type VulnerabilityMatcher struct {
	Store          store.Store
	Matchers       []matcher.Matcher
	IgnoreRules    []match.IgnoreRule
	FailSeverity   *vulnerability.Severity
	NormalizeByCVE bool
}
```

I just thought it will be nicer to pass it and not ignore it, currently it will be `nil` as before.